packages/wallets/src/features/cashier/modules/Transfer/components/TransferFormAmountInput/TransferFormAmountInput.tsx

```diff
@@ -0,0 +1,113 @@
+        : toAccount?.currencyConfig?.fractional_digits;
+    const isAmountInputDisabled = fieldName === 'toAmount' && !toAccount;
+
+    useEffect(() => {
+        if (!fromAccount?.currency || !toAccount?.currency || !activeWallet?.loginid) return;
+        subscribe({
+            base_currency: fromAccount?.currency,
+            loginid: activeWallet?.loginid,
+            target_currency: toAccount?.currency,
+        });
+        return () => unsubscribe();
+    }, [
+        activeWallet?.currency,
+        activeWallet?.loginid,
+        fromAccount?.currency,
+        subscribe,
+        toAccount?.currency,
+        unsubscribe,
+    ]);
+
+    const debouncedAmountConverterHandler = useMemo(
+        () =>
+            debounce((value: number) => {
+                if (!toAccount?.currency || !exchangeRate?.rates) return;
+
+                const toRate = exchangeRate.rates[toAccount.currency];
+
+                if (isFromAmountFieldName) {
+                    const convertedToAmount = Number(
+                        (value * toRate).toFixed(toAccount?.currencyConfig?.fractional_digits)
+                    );
+                    setFieldValue('toAmount', convertedToAmount);
+                } else {
+                    const convertedFromAmount = Number(
+                        (value / toRate).toFixed(fromAccount?.currencyConfig?.fractional_digits)
+                    );
+                    setTimeout(() => setFieldValue('fromAmount', convertedFromAmount));
+                }
+            }, 500),
+        [
+            exchangeRate?.rates,
+            fromAccount?.currencyConfig?.fractional_digits,
+            isFromAmountFieldName,
+            setFieldValue,
+            toAccount?.currency,
+            toAccount?.currencyConfig?.fractional_digits,
+        ]
+    );
+
+    const onChangeHandler = useCallback(
+        (value: number) => {
+            if (isCurrencyEqual) {
+                setValues(prev => ({ ...prev, fromAmount: value, toAmount: value }));
+            } else {
+                if (value === 0) {
+                    debouncedAmountConverterHandler.cancel();
+                    setValues(prev => ({ ...prev, fromAmount: 0, toAmount: 0 }));
+                    return;
+                }
+                setFieldValue(fieldName, value);
+                debouncedAmountConverterHandler(value);
+            }
+        },
+        [isCurrencyEqual, setValues, setFieldValue, fieldName, debouncedAmountConverterHandler]
+    );
+
+    const onFocusHandler = useCallback(() => {
+        setFieldValue('activeAmountFieldName', fieldName);
+    }, [fieldName, setFieldValue]);
+
+    return (
+        <ATMAmountInput
+            currency={currency}
+            disabled={isAmountInputDisabled}
+            fractionDigits={fractionDigits}
+            label={label}
+            maxDigits={MAX_DIGITS}
+            onChange={fieldName === values.activeAmountFieldName ? onChangeHandler : undefined}
```

Comment:
Here we are adding `onChangeHandler` to only one of the inputs.Because inputs should be synchronised, we will handle amount for both inputs in one handler. `onChangeHandler` will be attached to the input which is currently in focus state.

Reply:
Can u check from the `onChangeHandler` instead of dynamically attaching event handlers like this? See above. Doing this could cause certain elements to be non-interactive if u don't account for edge cases

Reply:
sorted